packages/salesforcedx-apex-debugger/src/adapter/apexDebug.ts

```diff
@@ -977,10 +987,56 @@ export class ApexDebug extends LoggingDebugSession {
         this.myRequestService.connectionTimeoutMs =
           workspaceSettings.connectionTimeoutMs;
         break;
+      case EXCEPTION_BREAKPOINT_REQUEST:
+        const requestArgs: SetExceptionBreakpointsArguments = args;
+        if (requestArgs && requestArgs.exceptionInfo) {
+          try {
+            await this.lock.acquire('exception-breakpoint', async () => {
+              return this.myBreakpointService.reconcileExceptionBreakpoints(
+                this.sfdxProject,
+                this.mySessionService.getSessionId(),
+                requestArgs.exceptionInfo
+              );
+            });
+            if (
+              requestArgs.exceptionInfo.breakMode ===
+              EXCEPTION_BREAKPOINT_BREAK_MODE_ALWAYS
+            ) {
+              this.printToDebugConsole(
+                nls.localize(
+                  'created_exception_breakpoint_text',
+                  requestArgs.exceptionInfo.label
+                )
+              );
+            } else if (
+              requestArgs.exceptionInfo.breakMode ===
+              EXCEPTION_BREAKPOINT_BREAK_MODE_NEVER
+            ) {
+              this.printToDebugConsole(
+                nls.localize(
+                  'removed_exception_breakpoint_text',
+                  requestArgs.exceptionInfo.label
+                )
+              );
+            }
+          } catch (error) {
+            response.success = false;
+            this.log(
+              TRACE_CATEGORY_BREAKPOINTS,
+              `exceptionBreakpointRequest: error=${error}`
+            );
+          }
+        }
+        break;
+      case LIST_EXCEPTION_BREAKPOINTS_REQUEST:
```

Comment:
This is a request from the extension to the adapter for a list of enabled exception breakpoints.